### ISSUE TITLE
Add a special case warning for parens in `for f in (a b)`

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#1229]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

It's somewhat common to try to put parens around for arguments:

```sh
for f in (a b c)
do echo "$f"; done
```

Examples:

```
for dir in ( cat foo.txt )
for file in ( file* )
for num in ( 0..100 )
```

#### Here's what shellcheck currently says:

Currently, this is caught with a generic "shell word starts with (" check:

```
for f in (a b c)
^-- SC1073: Couldn't parse this for loop. Fix to allow more checks.
         ^-- SC1036: '(' is invalid here. Did you forget to escape it?
```

But obviously it's not supposed to be escaped.

#### Here's what I wanted or expected to see:

Since the intent is known, a specific warning could be more helpful:

```
for f in (a b c)
^-- SC1073: Couldn't parse this for loop. Fix to allow more checks.
         ^-- SC1XXX: Remove invalid (..) around `for` arguments, or use `while read` to process command output.
```
